core/drivers/fabric-driver/server/events.ts

```diff
@@ -344,11 +348,77 @@ async function signEventSubscriptionQuery(
     }
 }
 
+async function writeExternalStateHelper(
+    writeExternalStateMessage: driverPb.WriteExternalStateMessage,
+    networkName: string
+) : Promise<any> {
+    const viewPayload: state_pb.ViewPayload = writeExternalStateMessage.getViewPayload();
+    const ctx: eventsPb.ContractTransaction = writeExternalStateMessage.getCtx();
+    const keyCert = await getDriverKeyCert();
+    let gateway: Gateway;
+
+    const requestId: string = viewPayload.getRequestId();
+    if (!viewPayload.getError()) {
+
+        let interopArgIndices = [], viewsSerializedBase64 = [], addresses = [], viewContentsBase64 = [];
+        const view: state_pb.View = viewPayload.getView();
+
+        const result = InteroperableHelper.getResponseDataFromView(view, keyCert.key.toBytes());
+        if (result.interopPayload.getConfidential()) {
+            viewContentsBase64.push(result.contents.toString("base64"));
+        } else {
+            viewContentsBase64.push("");
+        }
+
+        interopArgIndices.push(ctx.getReplaceArgIndex());
+        addresses.push(result.interopPayload.getAddress());
+        viewsSerializedBase64.push(Buffer.from(viewPayload.getView().serializeBinary()).toString("base64"));
+
+        let ccArgsB64 = ctx.getArgsList();
+        let ccArgsStr = [];
+        for (const ccArgB64 of ccArgsB64) {
+            ccArgsStr.push(Buffer.from(ccArgB64).toString('utf8'));
+        }
+
+        const invokeObject = {
+            channel: ctx.getLedgerId(),
+            ccFunc: ctx.getFunc(),
+            ccArgs: ccArgsStr,
+            contractName: ctx.getContractId()
+        }
+        console.debug(`invokeObject.ccArgs: ${invokeObject.ccArgs}`)
+
+        gateway = await getNetworkGateway(networkName);
+        const network: Network = await gateway.getNetwork(ctx.getLedgerId());
+        const interopContract: Contract = network.getContract(process.env.INTEROP_CHAINCODE ? process.env.INTEROP_CHAINCODE : 'interop');
+
+        const [ response, responseError ] = await handlePromise(InteroperableHelper.submitTransactionWithRemoteViews(
+            interopContract,
+            invokeObject,
+            interopArgIndices,
+            addresses,
+            viewsSerializedBase64,
+            viewContentsBase64
+        ));
+        if (responseError) {
+            console.log(`Failed writing to the ledger with error: ${responseError}`);
+            throw responseError;
```

Comment:
I think we need `gateway.disconnect` here before throwing error.

Reply:
Done